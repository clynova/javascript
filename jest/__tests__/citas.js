import Citas from '../js/classes/Citas'

describe('Probar la clase de citas', () => {

    const citas = new Citas()

    const id = Date.now()

    test('Agrega una nueva clase citas', () => {
        const citaObj = {
            id: id,
            mascota: 'Hook',
            propietario: 'TestPepe',
            telefono: '123123',
            fecha: '10-12-1993',
            hora: '10:30',
            sintomas: 'Solo duerme'
        }

        citas.agregarCita(citaObj)

        expect(citas).toMatchSnapshot()
    });

    test('Actualizar cita', () => {
        const citaObj = {
            id: id,
            mascota: 'Nuevo nombre',
            propietario: 'TestPepe',
            telefono: '123123',
            fecha: '10-12-1993',
            hora: '10:30',
            sintomas: 'Solo duerme'            
        }

        citas.editarCita(citaObj)

        expect(citas).toMatchSnapshot()
    });


    test('Eliminar cita', () => {
        citas.eliminarCita(id)

        expect(citas).toMatchSnapshot()
    });

});
