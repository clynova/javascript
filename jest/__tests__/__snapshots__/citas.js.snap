// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-1993",
      "hora": "10:30",
      "id": 1737419832245,
      "mascota": "Nuevo nombre",
      "propietario": "TestPepe",
      "sintomas": "Solo duerme",
      "telefono": "123123",
    },
  ],
}
`;

exports[`Probar la clase de citas Agrega una nueva clase citas 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-1993",
      "hora": "10:30",
      "id": 1737419832245,
      "mascota": "Hook",
      "propietario": "TestPepe",
      "sintomas": "Solo duerme",
      "telefono": "123123",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
